dotnet/src/SemanticKernel.Abstractions/Memory/MemoryQueryResult.cs

```diff
@@ -68,6 +53,7 @@ public class MemoryQueryResult
     {
         return new MemoryQueryResult(
             (MemoryRecordMetadata)rec.Metadata.Clone(),
-            relevance);
+            relevance,
+            rec.Embedding.IsEmpty ? null : rec.Embedding);
```

Comment:
can't you just use rec.Embedding here?

Reply:
This is due to the decision to keep MemoryRecord.Embedding non-nullable. It looks a little awkward here, but I would expect users to want Embedding vs Null instead of Embedding vs EmptyEmbedding.

Reply:
Discussed with @Vivihung offline